client/internal/clientcommon.go

```diff
@@ -257,10 +258,68 @@ func (c *ClientCommon) UpdateEffectiveConfig(ctx context.Context) error {
 	c.sender.NextMessage().Update(func(msg *protobufs.AgentToServer) {
 		msg.EffectiveConfig = cfg
 	})
+	// TODO: if this call is coming from OnMessage callback don't schedule the send
+	// immediately, wait until the end of OnMessage to send one message only.
 	c.sender.ScheduleSend()
 
 	// Note that we do not store the EffectiveConfig anywhere else. It will be deleted
 	// from NextMessage when the message is sent. This avoids storing EffectiveConfig
 	// in memory for longer than it is needed.
 	return nil
 }
+
+func (c *ClientCommon) SetRemoteConfigStatus(status *protobufs.RemoteConfigStatus) error {
+	if status.LastRemoteConfigHash == nil {
+		return errLastRemoteConfigHashNil
+	}
+
+	// Get the hash of the status before we update it.
+	prevHash := c.ClientSyncedState.RemoteConfigStatus().GetHash()
+
+	// Remember the new status.
+	if err := c.ClientSyncedState.SetRemoteConfigStatus(status); err != nil {
+		return err
+	}
+
+	// Check if the new status is different from the previous by comparing the hashes.
+	if !bytes.Equal(prevHash, status.Hash) {
+		// Let the Agent know about the new status.
```

Comment:
```suggestion
		// Let the Server know about the new status.
```

It seems that this comment should indicate that we're letting the Server know since we're about to send an AgentToServer message.

Reply:
Fixed.